hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -175,6 +219,57 @@ public void testWriteNullReadPrimitiveThrowsException() {
         assertReadNullAsPrimitiveThrowsException(serializedRecord);
     }
 
+    @Test
+    public void testWriteNullReadPrimitiveThrowsExceptionWithCorrectMethodPrefixCompactReader() {
+        SerializationService serializationService = createSerializationServiceWithASerializer();
+        SerializationService serializationService2 = createSerializationService();
+
+        // Reading null value with non-nullable reader method
+        A a1 = new A(null, new Integer[]{1, 2, 3});
+        Data data = serializationService.toData(a1);
+        // Reading compact with serializer case
+        errorMessageIncludes(assertThrows(HazelcastSerializationException.class, () -> serializationService.toObject(data)), "Use readNullable");
```

Comment:
maybe you could use `assertThat` here, for more readibility

Reply:
I guess you mean renaming `errorMessageIncludes` to `assertThatErrorMessageIncludes`